### ISSUE TITLE
question

### ISSUE THREAD

**Issue by dd33dd** [yonghanyu/DeepSearch#11]
Dear author, I would like to know whether the search engine DeepSearch is strictly an open search tool or a non-open search tool. I gave the --embedding instruction and its related address during the search. I specified the oxidation modification in advance during the compute embedding, and also added the --open-search instruction during the search process, and after the search was completed, the result was only found that the oxidation was modified, and no other modifications such as phosphorylation were found, and I added oxidation and phosphorylation at the same time as the compute embedding. The end result allows for the identification of both oxidation and phosphorylation. So when I do open-search, I can only find the modifiers that have been added to the embedding in advance, and the open search is not able to discover the modifiers outside the designation, so is deepsearch not strictly an open search tool?thank you.

**Comment by yonghanyu** [comment]
Hi there,

DeepSearch is for narrow-window search only at this moment. The open search feature is experimental and under development at this moment. Model trained for narrow-window search can not be applied directly in open search.

**Comment by dd33dd** [comment]
Thank you again for taking the time to reply despite your busy schedule. I would like to know what the accuracy and recall rate of amino acids can reach when I train the model. Currently, with my self-built dataset, after ten rounds of training, the accuracy is only about ten percent.

**Comment by yonghanyu** [comment]
It should reach around 0.9.

**Comment by dd33dd** [comment]
Thank you for your reply, author. Is my low accuracy during the training process caused by the following reasons? First, I made slight modifications to the code when running it. For example, here: train_one_epoch(model=model,epoch_index=epoch,train_loader=train_loader,train_sampler=train_sampler,loss_fn=loss,optimizer=optimizer,accum_freq=args.accum_freq,scaler=scaler,scheduler=schedular,device=device,args=args,tb_writer=tb_writer,tokenizer=tokenizer,) When I was training, it prompted that I had passed the tokenizer parameter incorrectly, so I removed it. Second, could it be an issue with the dataset I constructed? Because the dataset you previously shared gave the following error: Link iteration failed (wrong fractal heap header signature), which seems to indicate internal corruption. Therefore, I used my own constructed dataset (according to the required format). I look forward to your reply, thank you very much!
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yonghanyu/DeepSearch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年5月15日(星期四) 上午10:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yonghanyu/DeepSearch] question (Issue #11)



 yonghanyu left a comment (yonghanyu/DeepSearch#11)
 
It should reach around 0.9.

**Comment by yonghanyu** [comment]
I will take a look at the tokenizer. Besides, I checked the dataset, and it works fine with my setup. I suggest you check your h5py version and ensure you have the latest version installed.
